cmd/speccheck/spec.go

```diff
@@ -125,12 +128,12 @@ func checkCDOR(obj openrpc.ContentDescriptorOrReference) error {
 	return nil
 }
 
-func readSpec(path string) (*openrpc.OpenrpcDocument, error) {
+func readSpec(path string) (*ExtendedOpenrpcDocument, error) {
```

Comment:
For the spec or this reading of the spec,  I'm not sure if you'll have a dereferencer step already on the outside, that would make this easier as you wouldn't need to figure out how to resolve the json pointers, and so the ref types essentially wouldn't need to be specified. 

It would just be a matter of general correctness, that you could then throw an error on, if someone passes you a ref vs the object, in the additional places. It might be nice to have those checks on the deserialiization like you have for the methodobject

Overall I think this makes sense

Reply:
Yea the spec should be fully de-referenced before passing on to the speccheck by this https://github.com/simsonraj/execution-apis/blob/error_groups/scripts/build.js#L145

I have also added additional validations, as you mentioned in the earlier comments